server/migrations/versions/cec7ecc73bd8_samplesizeoptions_table.py

```diff
@@ -0,0 +1,63 @@
+# pylint: disable=invalid-name
+"""SampleSizeOptions table
+
+Revision ID: cec7ecc73bd8
+Revises: 496ee3db6da8
+Create Date: 2022-04-25 22:57:54.528446+00:00
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+# revision identifiers, used by Alembic.
+revision = "cec7ecc73bd8"
+down_revision = "496ee3db6da8"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table(
+        "sample_size_options",
+        sa.Column("created_at", sa.DateTime(), nullable=False),
+        sa.Column("updated_at", sa.DateTime(), nullable=False),
+        sa.Column("election_id", sa.String(length=200), nullable=False),
+        sa.Column("round_num", sa.Integer(), nullable=False),
+        sa.Column("task_id", sa.String(length=200), nullable=True),
+        sa.Column("sample_size_options", sa.JSON(), nullable=True),
+        sa.ForeignKeyConstraint(
+            ["election_id"],
+            ["election.id"],
+            name=op.f("sample_size_options_election_id_fkey"),
+            ondelete="cascade",
+        ),
+        sa.ForeignKeyConstraint(
+            ["task_id"],
+            ["background_task.id"],
+            name=op.f("sample_size_options_task_id_fkey"),
+            ondelete="set null",
+        ),
+        sa.PrimaryKeyConstraint(
+            "election_id", "round_num", name=op.f("sample_size_options_pkey")
+        ),
+    )
+
+    op.execute(
+        """
+        INSERT INTO sample_size_options
+            (created_at, updated_at, election_id, round_num, task_id, sample_size_options)
+        SELECT created_at, updated_at, id, 1, sample_size_options_task_id, sample_size_options
+        FROM election
+        """
+    )
```

Comment:
Nice, so this is migrating existing sample size data in the election table to the new dedicated sample size table. Is this necessary because we persist data from past audits in our databases? (Indefinitely?) Or do we need to do this because we have some in-flight audits right now?

Reply:
Currently we persist data until audit admins delete it. Setting an expiration date is an idea in our product backlog.

Migrations auto-run during the deploy process: https://devcenter.heroku.com/articles/release-phase#when-does-the-release-command-run. I generally try not to release during an active audit (especially for migrations), but even when I do it's usually pretty seamless.

Reply:
Nice, thanks for that context! And awesome to hear that even releasing during an active audit is pretty seamless